python/ray/data/_internal/batcher.py

```diff
@@ -323,3 +373,12 @@ def next_batch(self) -> Block:
         return BlockAccessor.for_block(self._shuffle_buffer).slice(
             slice_start, self._batch_head
         )
+
+
+def _get_total_obj_store_mem_on_node() -> int:
+    node_id = ray.get_runtime_context().get_node_id()
+    total_resources_per_node = ray._private.state.total_resources_per_node()
```

Comment:
IIRC this API requires an RPC. Since this is only called once per iteration, I think the performance should be good enough

Reply:
> IIRC this API requires an RPC

Why don't we move this into `DataContext` and cache it there?

Reply:
maybe put it in `data/_internal/util.py`, as it is a utility function